kube/stateful_set.go

```diff
@@ -86,12 +86,25 @@ func getVolumeClaims(role *model.InstanceGroup, createHelmChart bool) []helm.Nod
 		}
 
 		meta := helm.NewMapping("name", volume.Tag)
-		annotationList := helm.NewMapping()
-		annotationList.Add(VolumeStorageClassAnnotation, storageClass)
-		for key, value := range volume.Annotations {
-			annotationList.Add(key, value)
+		if len(volume.Annotations) > 0 {
+			annotationList := helm.NewMapping()
+			haveAnnotations := false
+			for key, value := range volume.Annotations {
+				// Detect and convert a storage-class annotation in the role-manifest to
+				// a regular storage class name. Needed because such an annotation is the
+				// only way for the role manifest to override the type-derived storage
+				// class, yet we must not emit this specific annotation anymore.
```

Comment:
Are we actually using this anywhere?

Reply:
I saw this in a unit test, and not having this code breaks that test. As such I assumed that this is something which can happen in the wild. 

See
`./test-assets/role-manifests/kube/volumes-with-annotation.yml:21:              volume.beta.kubernetes.io/storage-class: a-company-file-gold`

Reply:
"In the wild" essentially means in SCF or UAA, and I don't see them using it.

I would prefer to get rid of this "feature" instead of writing new code to implement it. If you decide to keep this code, then I think you also need to figure out what to do with the `storage-provisioner` annotation:

```
  annotations:
    volume.beta.kubernetes.io/storage-class: a-company-file-gold
    volume.beta.kubernetes.io/storage-provisioner: a-company.io/storage-provisioner
```

It seems inconsistent to transform one, but not the other. Not translating either of them seems easier. 😄

Reply:
Ok. Will remove that part, and update the unit test to match.

Reply:
Addressed with commit 0662aedf2624e45f8529df6d0c250d8196fe7902.